tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,490 @@
+                'dateformat': matching_dateformat[datetimeformat],
+                'datetimeformattz': datetimeformat,
+            }
+            uses_pandas = False
+            if self.engine.name == 'redshift':
+                if datetimeformat != 'YYYY-MM-DD HH24:MI:SS':
+                    # this is the only format supported by Redshift on
+                    # export
+                    uses_pandas = True
+            elif self.engine.name == 'mysql':
+                # mysql has no exporter defined, so everything is via Pandas
+                uses_pandas = True
+            elif self.engine.name == 'postgresql':
+                if datetimeformat in ['YYYY-MM-DD HH:MI:SS',
+                                      'YYYY-MM-DD HH24:MI:SS']:
+                    # ensure we keep other areas of ISO style
+                    addl_hints = {
+                        'timeonlyformat': 'HH:MI:SS',
+                        'dateformat': 'YYYY-MM-DD',
+                        'datetimeformattz': 'YYYY-MM-DD HH:MI:SSOF',
+                    }
+                else:
+                    uses_pandas = True
+            elif self.engine.name == 'vertica':
+                # Make sure our '\n' strings below are valid when comparing output
+                addl_hints['record-terminator'] = '\n'
+                if datetimeformat not in ['YYYY-MM-DD HH:MI:SS',
+                                          'YYYY-MM-DD HH24:MI:SS']:
+                    uses_pandas = True
+            if uses_pandas:
+                # TODO: Should I verify this?
+                # We're going to need to be sure to use hints that
+                # work in Pandas
+                addl_hints.update(pandas_compatible_addl_hints)
+                if 'AM' in datetimeformat:
+                    # TODO: Add a GitHub issue for this
+                    # TODO: Should expect failure and complain if it doesn't fail
+                    logger.warning('Cannot export this dateformat using Pandas--'
+                                   'skipping test')
+                    continue
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'datetimeformat': datetimeformat,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            expect_pandas_failure = not self.has_pandas() and uses_pandas
+            try:
+                csv_text = self.unload(column_name='timestamp',
+                                       records_format=records_format)
+                self.assertIn(csv_text, [create_sample(datetimeformat) + "\n",
+                                         # TODO: Should this be necessary?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker